en/services/parameter.md

```diff
@@ -79,33 +84,74 @@ sequenceDiagram;
     Drone->>Drone: Start sending parameters 
     Drone->>GCS: Send N parameters with PARAM_VALUE
     GCS->>GCS: Start receive timeout
-    GCS->>Drone: If some params are dropped, request with PARAM_REQUEST_READ
+    GCS-->>Drone: Request any dropped params with PARAM_REQUEST_READ
+{% endmermaid %}
+
+The sequence of operations is:
+
+1. GCS (client) sends [PARAM_REQUEST_LIST](../messages/common.md#PARAM_REQUEST_READ) specifying the target system/component.
+1. Drone sends all parameters individually in [PARAM_VALUE](../messages/common.md#PARAM_VALUE) messages.
+   - The drone should allow a break between each message in order to avoid saturating the link.
+1. GCS accumlates parameters in order to know which parameters have been/not been received  (`PARAM_VALUE` contains total number of params and index of current param).
+1. GCS starts timeout after each `PARAM_VALUE` message in order to detect when parameters are no longer being sent.
+1. After timeout (messages no longer being sent) the GCS can request any missing parameter values by [requesting them individually](#read_single) (using [PARAM_REQUEST_READ](../messages/common.md#PARAM_REQUEST_READ)).
+
+
+
+### Read Single Parameter {#read_single}
+
+A single parameter can be read by sending the [PARAM_REQUEST_READ](../messages/common.md#PARAM_REQUEST_READ) message, as shown below:
+
+{% mermaid %}
+sequenceDiagram;
+    participant GCS
+    participant Drone
+    GCS->>Drone: PARAM_REQUEST_READ
+    GCS->>GCS: Start receive timeout
+    Drone->>GCS: PARAM_VALUE
+    GCS-->>Drone: Re-request parameter value on timeout
 {% endmermaid %}
 
-### Read Single Parameter
+The sequence of operations is:
 
-A single parameter can be read by the [PARAM_REQUEST_READ](../messages/common.md#PARAM_REQUEST_READ) message.
+1. GCS (client) sends [PARAM_REQUEST_READ](../messages/common.md#PARAM_REQUEST_READ) specifying the either the parameter name or index.
+1. GCS starts timout waiting for acknowledgement (in the form of a [PARAM_VALUE](../messages/common.md#PARAM_VALUE) message).
+1. Drone responds with `PARAM_VALUE` containing the parameter value.
+   This is a broadcast message (sent to all systems).
 
+The drone may restart the sequence the `PARAM_VALUE` acknowledgement is not received within the timeout.
 
 
-### Write Parameters
+### Write Parameters {#write}
```

Comment:
Docs on PARAM_SET say:

> Set a parameter value TEMPORARILY to RAM. It will be reset to default on system reboot. Send the ACTION MAV_ACTION_STORAGE_WRITE to PERMANENTLY write the RAM contents to EEPROM. ...

1. Is this really only temporarily stored? If so then this needs to be said here (otherwise the param itself needs to be updated).
2. I can't find MAV_ACTION_STORAGE_WRITE - please clarify.

Reply:
@julianoes Any thoughts on this one?

Reply:
Fixed by https://github.com/mavlink/mavlink/pull/1018